### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by Ronningen**

### CODE REVIEW

File: term_1/solver.cpp [hunk 1]
```diff
@@ -133,8 +133,7 @@ struct KahanVector // Вектор переопределяющий сумму 
 private:
     void collapse()
     {
-        // segmentation fault if value.size() == 0
-        for (auto v = value.begin() + 1; v != value.end(); v++)
+        for (auto v = value.begin() + 1; v != value.end(); v++) // v.size() != 0
```
Comment (csintroformipt) [inline term_1/solver.cpp]: validation of v..size is required in case the code crashes in this case

**Review by csintroformipt — APPROVED** [review]
Smart idea to use customization parameters for stepping (constraints) and for results output.
Currently the upper level (parse and run) is somewhat useless as there is only one routine in main() yet the idea is nice. As a further imrovement i'd suggest to add functionality to proceed the with the simulation from the existing timestep (to enable lauchhing a simulation to time t=t1 and then continuing simulation from time t1 to time t2 with different parameters).
Nice job!

File: term_1/solver.cpp (hunk 1 above)
Comment (Ronningen) [reply term_1/solver.cpp]: Thanks. Well, v.size() == 0 is logical error and each constructor creates non-empty vector. I can throw logic_error exception in this point, but i think that several validations in every step of calculations will negatively effect the efficiency.

Comment (csintroformipt) [reply term_1/solver.cpp]: Sure. Validations should be done either upon or during the configuring step. I didn't notice that the vectors there was non-empty and was guided mainly by your comments. Adding any run-time checks in core simulator code is definitely not a wise idea in any case. Even for the sake of debugging the asserts would be sufficient (and would not cause any additional operations in release builds).

**Pull request merged**